xarray/core/indexes.py

```diff
@@ -1800,7 +1844,10 @@ def isel_indexes(
     indexes: Indexes[Index],
     indexers: Mapping[Any, Any],
 ) -> tuple[dict[Hashable, Index], dict[Hashable, Variable]]:
-    return _apply_indexes(indexes, indexers, "isel")
+    if any(isinstance(v, PandasMultiIndex) for v in indexes._indexes.values()):
```

Comment:
FYI: this is the error that shows up without this if statement:
<details>

```python
    def _assert_indexes_invariants_checks(
        indexes, possible_coord_variables, dims, check_default=True
    ):
        assert isinstance(indexes, dict), indexes
        assert all(isinstance(v, Index) for v in indexes.values()), {
            k: type(v) for k, v in indexes.items()
        }

        index_vars = {
            k for k, v in possible_coord_variables.items() if isinstance(v, IndexVariable)
        }
        assert indexes.keys() <= index_vars, (set(indexes), index_vars)

        # check pandas index wrappers vs. coordinate data adapters
        for k, index in indexes.items():
            if isinstance(index, PandasIndex):
                pd_index = index.index
                var = possible_coord_variables[k]
                assert (index.dim,) == var.dims, (pd_index, var)
                if k == index.dim:
                    # skip multi-index levels here (checked below)
                    assert index.coord_dtype == var.dtype, (index.coord_dtype, var.dtype)
                assert isinstance(var._data.array, pd.Index), var._data.array
                # TODO: check identity instead of equality?
                assert pd_index.equals(var._data.array), (pd_index, var)
            if isinstance(index, PandasMultiIndex):
                pd_index = index.index
                for name in index.index.names:
                    assert name in possible_coord_variables, (pd_index, index_vars)
                    var = possible_coord_variables[name]
                    assert (index.dim,) == var.dims, (pd_index, var)
                    assert index.level_coords_dtype[name] == var.dtype, (
                        index.level_coords_dtype[name],
                        var.dtype,
                    )
                    assert isinstance(var._data.array, pd.MultiIndex), var._data.array
                    assert pd_index.equals(var._data.array), (pd_index, var)
                    # check all all levels are in `indexes`
                    assert name in indexes, (name, set(indexes))
                    # index identity is used to find unique indexes in `indexes`
>                   assert index is indexes[name], (pd_index, indexes[name].index)
E                   AssertionError: (MultiIndex([('a', 1, -1),
E                               ('b', 2, -2)],
E                              name='x'), MultiIndex([('a', 1, -1),
E                               ('b', 2, -2)],
E                              name='x'))

/home/mark/git/xarray/xarray/testing/assertions.py:411: AssertionError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/mark/git/xarray/xarray/testing/assertions.py(411)_assert_indexes_invariants_checks()
-> assert index is indexes[name], (pd_index, indexes[name].index)
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>


======================================== short test summary info =========================================
FAILED xarray/tests/test_dataarray.py::TestDataArray::test_selection_multiindex - AssertionError: (MultiIndex([('a', 1, -1),
```

</details>

Reply:
looks like a copy was made?

Reply:
yeah, i just don't know enough about the internal datastructures of xarray and ran out of time hunting for it.

Maybe there is a different way of iterating through things.

Reply:
Mind adding a `TODO: remove `if` clause in the future. It should be unnecessary.` commment please

Reply:
added